scale/scheduler/resources/manager.py

```diff
@@ -285,8 +290,25 @@ def sync_with_mesos(self, host_address):
         resources = {}
         try:
             resources = get_agent_resources(host_address, agents_needing_totals)
-        except:
-            logger.exception('Error getting agent resource totals from Mesos')
+        except KeyError as ex:
+            logger.exception('Error getting agent resource totals from Mesos: missing key %s' % ex)
+            self._mesos_error = 'Missing key %s in mesos response' % ex
+            if not self._mesos_error:
+                self._mesos_error_started = datetime.now()
+        except Exception as ex:
+            logger.exception('Error getting agent resource totals from Mesos: %s' % ex)
+            self._mesos_error = ex.message
+            if not self._mesos_error:
+                self._mesos_error_started = datetime.now()
+
+        if self._mesos_error_started and datetime.now() > self._mesos_error_started + SHUTDOWN_PERIOD:
+            from scheduler.management.commands.scale_scheduler import GLOBAL_SHUTDOWN
```

Comment:
Lets log either here or in GLOBAL_SHUTDOWN (_shutdown in scale_scheduler) that we are forcibly going down.

Reply:
After thinking about it lets log here the reason for the shutdown.